producer-services/src/main/scala/za/co/absa/spline/producer/service/repo/ExecutionProducerRepository.scala

```diff
@@ -24,4 +24,5 @@ trait ExecutionProducerRepository {
   def insertExecutionPlan(executionPlan: ExecutionPlan)(implicit ec: ExecutionContext): Future[Unit]
   def insertExecutionEvents(executionEvents: Array[ExecutionEvent])(implicit ec: ExecutionContext): Future[Unit]
   def isDatabaseOk()(implicit ec: ExecutionContext): Future[Boolean]
+  def getDataSources(value: String, access: Optional[String])(implicit ec: ExecutionContext): Future[Array[String]]
```

Comment:
```suggestion
  def getDataSources(value: String, access: Option[String])(implicit ec: ExecutionContext): Future[Array[String]]
```
- Optional is Java class
- Option is Scala class

Reply:
making the change